src/sampleworks/eval/generate_synthetic_density.py

```
@@ -275,6 +320,21 @@ def _process_single_row(
         return
 
+    if save_structure:
+        # Shift coordinates into the grid frame so the saved CIF aligns with
+        # the CCP4 map. CCP4 format (unlike MRC) cannot encode an arbitrary Cartesian
+        # origin, so we move the atoms instead. Possible the better way is to resample the map?
+        atom_array.coord = atom_array.coord - xmap_torch.origin  # pyright: ignore[reportOptionalOperand]
+        structure_output_path = structure_path.parent / f"{structure_path.stem}_density_input.cif"
+        try:
+            save_structure_to_cif(atom_array, structure_output_path)
+            logger.info(f"Saved processed structure to {structure_output_path}")
+        except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)